dartagnan/src/main/java/com/dat3m/dartagnan/program/event/EventFactory.java

```diff
@@ -93,7 +94,15 @@ public static List<Event> eventSequence(Object... events) {
 
     public static Alloc newAlloc(Register register, Type allocType, Expression arraySize,
                                  boolean isHeapAlloc, boolean doesZeroOutMemory) {
-        return new Alloc(register, allocType, arraySize, isHeapAlloc, doesZeroOutMemory);
+        final Expression defaultAlignment = expressions.makeValue(8, types.getArchType());
+        return newAlignedAlloc(register, allocType, arraySize, defaultAlignment, isHeapAlloc, doesZeroOutMemory);
+    }
+
+    public static Alloc newAlignedAlloc(Register register, Type allocType, Expression arraySize, Expression alignment,
+                                 boolean isHeapAlloc, boolean doesZeroOutMemory) {
+        arraySize = expressions.makeCast(arraySize, types.getArchType(), false);
```

Comment:
Why do we now enforce `arraySize` to be 64 bits, but before it was not the case?

Reply:
We didn't enforce this because the size was assumed to be static and the encoding used the static value, totally ignoring its type. This made type-mismatches irrelevant. Now we use the proper expression and thus rely on correct types.
There is an argument to be made what type a size should be, but I think it is most canonical if its type matches with pointers (i.e., if pointers are 64 bit then sizes are 64 bit).